libs/entry-components/styles/modules/components/slide-togglers/_generator.scss

```diff
@@ -0,0 +1,33 @@
+/* stylelint-disable scss/selector-no-redundant-nesting-selector */
+@use 'sass:map';
+@mixin generate-from($theme) {
+	.mat-mdc-slide-toggle {
+		$enabled-states: 'enabled', 'enabled:focus:not(:active)', 'enabled:active', 'enabled:hover:not(:focus):not(:active)';
+
+		$track-on-background: map.get($theme, 'general', 'slide-togglers', 'on', 'track', 'background');
+		$track-off-background: map.get($theme, 'general', 'slide-togglers', 'off', 'track', 'background');
+		$handle-on-background: map.get($theme, 'general', 'slide-togglers', 'on', 'handle', 'background');
+		$handle-off-background: map.get($theme, 'general', 'slide-togglers', 'off', 'handle', 'background');
+
+		@each $state in $enabled-states {
```

Comment:
We want to make things configurable, everything fine.
I'm thinking whether we can configure single color or just two colors instead of four? What are the use cases?

In AM we have two colors and two lighter/darker variations of colors for track depending on lightness. Is it the same on our projects?
Can we define colors just for handle on/off and then calculate color for track which is 10% darker or lighter depending on lightness of given color?

Reply:
Good point, I agree that is better to get rid of too specific customization for now. Now we will have just on/off for handle. I created function that conditionally increases or decreases the lightness of a color based on the color's current lightness, unless the color is undefined or transparent.